cdk_framework/EKS/lib/stacks/ec2-cluster-stack.ts

```diff
@@ -32,40 +33,32 @@ export class ClusterStack extends Stack {
         ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy')
       ]
     });
-    if(props.launchType['ec2'] !== undefined){
-      this.cluster = new eks.Cluster(this, props.name, {
-        clusterName: props.name,
-        vpc: props.vpc,
-        vpcSubnets: [{subnetType: ec2.SubnetType.PUBLIC}],
-        defaultCapacity: 0,  // we want to manage capacity our selves
-        version: props.version,
-        clusterLogging: logging,
-      
-      });
-      const instanceType = props.launchType['ec2']['ec2_instance']
-      const instanceSize = props.launchType['ec2']['node_size']
-      this.cluster.addNodegroupCapacity('ng-' + instanceType, {
-          instanceTypes: [new ec2.InstanceType(instanceType + '.' + instanceSize)],
-          minSize: 2,
-          nodeRole: workerRole
-      })
-    } else if (props.launchType['fargate'] !== undefined){
-      this.cluster = new eks.FargateCluster(this, props.name, {
-        clusterName: props.name,
-        vpc: props.vpc,
-        version: props.version,
-        clusterLogging: logging
-      });
-    } 
+
+    const instance_type = props.instance_type.toLowerCase()
+
+    this.cluster = new eks.Cluster(this, props.name, {
+      clusterName: props.name,
+      vpc: props.vpc,
+      vpcSubnets: [{subnetType: ec2.SubnetType.PUBLIC}],
+      defaultCapacity: 0,  // we want to manage capacity our selves
+      version: props.version,
+      clusterLogging: logging,
     
+    });
+    this.cluster.addNodegroupCapacity(`ng-${instance_type}`, {
+        instanceTypes: [new ec2.InstanceType(instance_type)],
+        minSize: 2,
+        nodeRole: workerRole
+    })
     this.cluster.awsAuth.addMastersRole(Role.fromRoleName(this, 'eks_admin_role', 'Admin'))
+    this.cluster.awsAuth.addMastersRole(Role.fromRoleName(this, 'he', 'Admin'))
```

Comment:
What is the purpose of this master role?

Reply:
Oh shoot. That was a mistake, I was playing with things. Good catch!